willow/image.py

```diff
@@ -135,6 +135,7 @@ def has_animation(self):
 
 class ImageFile(Image):
     format_name = None
+    mime_type = None
```

Comment:
question: wouldn't `application/octet-stream` make sense for an unknown type

[RFC 2046](http://www.rfc-editor.org/rfc/rfc2046.txt) states in section 4.5.1:

> The "octet-stream" subtype is used to indicate that a body contains arbitrary binary data.

Reply:
I think mime_type should be required in the ImageFile implementation. Not assigning it should generally create an error.  Maybe I should reimplement this as a getter that throws and exception by default?

Reply:
- now raises a NotImplementedError if mime_type or format_name are missing from an ImageFile implementation. 
- added test coverage